aiida_lammps/workflows/base.py

```diff
@@ -0,0 +1,351 @@
+        2. Use a restartfile found in the remote folder from the previous calculation.
+        3. Use the structure from the last step of the trajectory from the previous calculation.
+        4. Restart from scratch
+        """
+        self.report("Walltime reached attempting restart")
+
+        latest_file = self._check_restart_in_remote(calculation=calculation)
+        if "restartfile" in calculation.outputs:
+            self.set_restart_type(
+                restart_type=RestartTypes.FROM_RESTARTFILE,
+                calculation=calculation,
+            )
+            self.report_error_handled(
+                calculation,
+                "restarting from the stored restartfile",
+            )
+        elif latest_file is not None:
+            self.set_restart_type(
+                restart_type=RestartTypes.FROM_REMOTEFOLDER,
+                calculation=calculation,
+            )
+            self.report_error_handled(
+                calculation,
+                "restarting from the remote folder of the previous calculation",
+            )
+        elif "trajectory" in calculation.outputs:
+            self.set_restart_type(
+                restart_type=RestartTypes.FROM_STRUCTURE,
+                calculation=calculation,
+            )
+            self.report_error_handled(
+                calculation,
+                "restarting from the last step of the trajectory",
+            )
+        else:
+            self.set_restart_type(
+                restart_type=RestartTypes.FROM_SCRATCH,
+                calculation=calculation,
+            )
+            self.report_error_handled(
+                calculation,
+                "restarting from scratch and increasing the walltime",
+            )
+        return ProcessHandlerReport(True)
+
+    @process_handler(
+        priority=620,
+        exit_codes=[
+            LammpsBaseCalculation.exit_codes.ERROR_FORCE_NOT_CONVERGED,  # pylint: disable=no-member
+            LammpsBaseCalculation.exit_codes.ERROR_ENERGY_NOT_CONVERGED,  # pylint: disable=no-member
+        ],
+    )
+    def handle_minimization_not_converged(self, calculation):
+        """
+        Handle calculations where the minimization did not converge
+
+        The handler will try to find a configuration to restart from with the
+        following priority
+
+        1. Use a stored restart file in the repository from the previous calculation.
+        2. Use a restartfile found in the remote folder from the previous calculation.
+        3. Use the structure from the last step of the trajectory from the previous calculation.
+
+        This handler should never start from restart as at least the trajectory
+        should always exist, if the calculation finished successfully.
+        """
+
+        if calculation.exit_status == 401:
+            self.report("Energy not converged during minimization, attempting restart")
+        if calculation.exit_status == 402:
```

Comment:
Probably best to use `LammpsBaseCalculation.exit_codes.ERROR_FORCE_NOT_CONVERGED.status` instead of hard-coding the numbers, just in case they ever were to change

Reply:
Sure thing will change it to that.